react-app/src/features/welcome/index.tsx

```diff
@@ -12,31 +12,28 @@ import { useFeatureFlag } from "@/hooks/useFeatureFlag";
 
 export const loader = (queryClient: QueryClient) => {
   return async () => {
+    // Check user query
+    const userFetch = await queryClient.fetchQuery({
+      queryKey: ["user"],
+      queryFn: () => getUser(),
+    });
+    // Verify location point if logged out
+    const noAuthPage = ["/login", "/faq"].includes(window.location.pathname);
```

Comment:
```ts
const shouldRedirect = !["/login", "/faq"].includes(window.location.pathname) && !user?.user;

if (shouldRedirect) {
  return <Navigate to="/login" />
}
```

Building on your excellent changes, but my naming is missing context, so perhaps you can fill that in!

Reply:
Looks great! Thanks for bearing with me. Looking at your work again...